spring-boot-project/spring-boot-tools/spring-boot-buildpack-platform/src/main/java/org/springframework/boot/buildpack/platform/docker/configuration/DockerConfiguration.java

```diff
@@ -63,16 +63,17 @@ public DockerRegistryAuthentication getPublishRegistryAuthentication() {
 		return this.publishAuthentication;
 	}
 
-	public DockerConfiguration withHost(String address, boolean secure, String certificatePath) {
+	public DockerConfiguration withHost(String address, boolean secure, String certificatePath, Integer socketTimeout) {
```

Comment:
This is a breaking API change. A separate method for the timeout would be better.